src/main/python/ayab/signal_receiver.py

```diff
@@ -55,25 +59,19 @@ class SignalReceiver(QObject):
     hw_test_starter = Signal(Control)
     hw_test_writer = Signal(str)
 
-    def __init__(self):
+    def __init__(self)->None:
         super().__init__()
 
-    def signals(self):
-        """Iterator over names of signals."""
-        return filter(
-            lambda x: type(getattr(self, x)).__name__ == "SignalInstance",
-            SignalReceiver.__dict__.keys())
-
-    def activate_signals(self, parent):
+    def activate_signals(self, parent:GuiMain)->None:
         self.start_row_updater.connect(parent.update_start_row)
         self.progress_bar_updater.connect(parent.progbar.update)
-        self.knit_progress_updater.connect(parent.knitprog.update)
+        self.knit_progress_updater.connect(parent.knitprog.update_progress)
         self.notifier.connect(parent.notify)
         # self.statusbar_updater.connect(parent.statusbar.update)
         self.blocking_popup_displayer.connect(display_blocking_popup)
         self.popup_displayer.connect(display_blocking_popup)
         self.audio_player.connect(parent.audio.play,
-                                  type=Qt.BlockingQueuedConnection)
+                                  type=Qt.ConnectionType.BlockingQueuedConnection) #type: ignore
```

Comment:
`type` kwarg expects a `type` but this is an `Enum` value. I'm not sure if the typing is wrong or not- [the Qt docs are half-baked at best](https://doc.qt.io/qtforpython-6/PySide6/QtCore/Signal.html#PySide6.QtCore.Signal.connect)

Reply:
The official Qt Types properly type this